05/DB.test.js

```diff
@@ -1 +1,111 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe ('Database', ()=> {
+    describe ('.insert()', () => {
+        it('should return inserted object with correct id', ()=> {
+            const database = new DB();
+            const promise = database.insert({a:3, b:2});
+            return promise
+                .then(()=> database.getRows())
+                .then(rows => {
+                    expect(rows.length).toBe(1)
+                }) 
+        })
+    
+        it('should throw proper error message if id is not a number', () => {
+            expect.assertions(1);
+            const database = new DB();
+            const wrongIdObject = { id: 'abc' }
+            return database.insert(wrongIdObject)
+              .catch(error => {
+                expect(error).toBe('ID can be only number!');
+              });
+        });
+    
+        it('should throw proper error message if if inserted id already exists in this._rows', () => {
+            const database = new DB();
+            const duplicatedIdObject = {id:1}
+            database._rows.push(duplicatedIdObject);
```

Comment:
Jeśli mamy pole prywatne (a to sugeruje podkreślnik tj. _) to nie powiniśmy korzystać z tego rozwiązania poza klasą.
Lepiej byłoby użyć "insert()" dwa razy.